codegen/src/main/ops/org/nd4j/codegen/ops/SDBaseOps.kt

```diff
@@ -0,0 +1,1806 @@
+        Arg(NUMERIC, "value") { description = "Scalar value to compare" }
+        Output(NUMERIC, "output"){ description = "Output variable" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Element-wise scalar floor modulus operation: out = floorMod(in, value).
+                i.e., returns the remainder after division by 'value'
+            """.trimIndent()
+        }
+    }
+
+    Op("scalarMax") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.scalar"
+        legacy = true
+        Input(NUMERIC, "in") { description = "Input variable" }
+        Arg(NUMERIC, "value") { description = "Scalar value to compare" }
+        Output(NUMERIC, "output"){ description = "Scalar value to compare" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Element-wise scalar maximum operation: out = max(in, value)
+            """.trimIndent()
+        }
+    }
+
+    Op("scalarMin") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.scalar"
+        legacy = true
+        Input(NUMERIC, "in") { description = "Input variable" }
+        Arg(NUMERIC, "value") { description = "Scalar value to compare" }
+        Output(NUMERIC, "output"){ description = "Output variable" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Element-wise scalar minimum operation: out = min(in, value)
+            """.trimIndent()
+        }
+    }
+
+    Op("scalarSet") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.scalar"
+        legacy = true
+        Input(NUMERIC, "in") { description = "Input variable" }
+        Arg(NUMERIC, "set") { description = "Value to set" }
+        Output(NUMERIC, "output"){ description = "Output variable" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Return a variable with equal shape to the input, but all elements set to value 'set'
+            """.trimIndent()
+        }
+    }
+
+    Op("scatterAdd") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.scatter"
+        Input(NUMERIC, "ref") { description = "Initial/source variable" }
+        Input(NUMERIC, "indices") { description = "Indices array" }
+        Input(NUMERIC, "updates") { description = "Updates to add to the initial/source array" }
+        Output(NUMERIC, "output"){ description = "The updated variable" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Scatter addition operation.
+                If indices is rank 0 (a scalar), then out[index, ...] += updates[...]
+                If indices is rank 1 (a vector), then for each position i, out[indices[i], ...] += updates[i, ...]
+                If indices is rank 2+, then for each position (i,...,k), out[indices[i], ..., indices[k], ...] += updates[i, ..., k, ...] 
+                Note that if multiple indices refer to the same location, the contributions from each is handled correctly.
+            """.trimIndent()
+        }
+    }
```

Comment:
Let's define a mixin for the scatter* ops - java package, inputs and output (i.e., everything except doc)

Reply:
Done. Also added a mixin for the identical part of the scatter Docs.